pkg/network/dhcp/dhcp.go

```diff
@@ -38,6 +60,61 @@ func (d *Probe) Start(inf string) error {
 	return nil
 }
 
+// Start runs the DHCP client process and registers it to zinit
+func (d *BackgroundProbe) Start() error {
+	serviceName := fmt.Sprintf("dhcp-%s", d.inf)
+
+	ns, err := exec.Command("ip", "netns", "identify").Output()
+	if err != nil {
+		return errors.Wrap(err, "failed to identify namespace")
+	}
+
+	exec := fmt.Sprintf("/sbin/udhcpc -v -f -i %s -t 20 -T 1 -s /usr/share/udhcp/simple.script", d.inf)
+
+	if strings.Trim(string(ns), " ") != "" {
+		exec = fmt.Sprintf("ip netns exec %s %s", ns, exec)
+	}
+
+	err = zinit.AddService(serviceName, zinit.InitService{
+		Exec:    exec,
+		Oneshot: false,
+		After:   []string{},
+	})
+
+	if err != nil {
+		log.Error().Err(err).Msg("fail to create dhcp-zos zinit service")
+		return err
+	}
+
+	if err := d.z.Monitor(serviceName); err != nil {
+		log.Error().Err(err).Msg("fail to start monitoring dhcp-zos zinit service")
+		return err
+	}
+
+	return nil
+}
+
+// IsRunning checks if a background process is running in zinit
+func (d *BackgroundProbe) IsRunning() (bool, error) {
+	serviceName := fmt.Sprintf("dhcp-%s", d.inf)
+
+	services, err := d.z.List()
+	if err != nil {
+		log.Error().Err(err).Msg("fail to create dhcp-zos zinit service")
```

Comment:
Wrong log message! Also no point of logging here, just make sure to wrap your err with the correct message.